### PR TITLE
Typescript conversion: page interface module

### PR DESCRIPTION
**Pull request by jahorton**
This PR serves to refactor and reorganize the code that originally corresponded to keymanweb.js.  It's a big file, so there are a lot of affected lines.

In order to facilitate testing, the newly refactored source must exist in separate files from the unrefactored source.  Otherwise, TypeScript starts running into problematic issues, as it wants to include unrefactored methods before they can be set.  (When it includes source through the `/// <reference/>` tag, it inlines the whole file.)

Note that there is periodic use of `(<any>)` casting as a placeholder for future refactoring work/targets; this permits testing while allowing TS to compile with the intermediate changes.

Also note that many of the API functions based on the original object have been preserved within the KeymanBase object using Closure-friendly syntax.  TypeScript plays nice with that coding style, and those functions exist solely as wrappers to the new internal API endpoints.

Four new code files exist:
* kmwdomevents.ts - Contains the DOM-related event handlers (at present, the touch-based event handler objects).  Also contains a number of event-related state variables that used to be under the base object.
* kmwdom.ts - Contains the DOM-related management code in a new `DOMManager` class; this class references the event-handler management objects from kmwdomevents.ts for managing interactivity.
    * Manages attachment data, handles DOM-related KMW API calls, and will contain the KMW init function.
* kmwhotkeys.ts - Contains hotkey-related code.  It's a small file, but it felt best as its own logical unit.
* kmwuimanager.ts - Contains UI management (and event generation) code.  Also a small file that didn't fit logically in any of the other files.

Note that this PR uses #504 as a base; significant parts of the refactored code refer to its changes.

### CODE REVIEW

**Comment by jahorton** [comment]
Note - `activeElement` and `lastActiveElement`, as seen in `CommonDOMStates`, are currently typeless due to issues referenced in #539.  Adding type will necessitate bug fixing, so this has been separated out as its own issue and will receive a separate PR.

File: web/source/kmwuimanager.ts [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  setActivatingUI(state: boolean|number) {
+    this.isActivating = state ? true : false;
+  }
+
+  /**
+   * Function     doLoad
+   * Scope        Private
+   * @return      {boolean}   
+   * Description  Execute UI initialization code after loading the UI
+   *              // Appears to be unused; could be eliminated?  Though, doUnload IS used.
```
Comment (mcdurdin) [inline web/source/kmwuimanager.ts]: Currently unused, but there so that other UI plugins could hook in if needed

Comment (jahorton) [reply web/source/kmwuimanager.ts]: It is... but by unused, I mean that KMW never raises the event/calls this method.  If a UI plugin tried to hook in, it'd get nothing presently.

**Comment by jahorton** [comment]
Yeah, I don't find that surprising.  When doing the Swift reviews, this scale probably took me about a day to fine-tooth-comb.

I've been making sure to do, at minimum, moderate testing as I progress through developing the PRs, testing with a diverse set of keyboards that seem to cover the general use cases.  I'll see about possibly developing 'engine' test cases in a PR soon.

**Pull request merged**